packages/react-charting/src/utilities/utilities.ts

```diff
@@ -1004,6 +1005,85 @@ export function domainRangeOfVSBCNumeric(
     : { dStartValue: xMin, dEndValue: xMax, rStartValue: rMax, rEndValue: rMin };
 }
 
+/**
+ * Calculates Domain and range values for Date X axis.
+ * This method calculates Bar chart.
+ * @export
+ * @param {IVerticalBarChartDataPoint[]} points
+ * @param {IMargins} margins
+ * @param {number} width
+ * @param {boolean} isRTL
+ * @param {Date[] | number[]} tickValues
+ * @returns {IDomainNRange}
+ */
+export function domainRangeOfDateForVerticalBarChart(
+  points: IVerticalBarChartDataPoint[],
+  margins: IMargins,
+  width: number,
+  isRTL: boolean,
+  tickValues: Date[] = [],
+  barWidth: number,
+): IDomainNRange {
+  const sDate = d3Min(points, (point: IVerticalBarChartDataPoint) => {
+    return point.x as Date;
+  })!;
+  const lDate = d3Max(points, (point: IVerticalBarChartDataPoint) => {
+    return point.x as Date;
+  })!;
+
+  // Need to draw graph with given small and large date (Which Involves customization of date axis tick values)
+  // That may be Either from given graph data or from prop 'tickValues' date values.
+  // So, Finding smallest and largest dates
+  //const smallestDate = d3Min([...tickValues, sDate])!;
+  //const largestDate = d3Max([...tickValues, lDate])!;
+
+  const rStartValue = margins.left! + barWidth / 2;
+  const rEndValue = width - margins.right! - barWidth / 2;
+
+  return isRTL
+    ? { dStartValue: lDate, dEndValue: sDate, rStartValue, rEndValue }
+    : { dStartValue: sDate, dEndValue: lDate, rStartValue, rEndValue };
+}
+
+/**
+ * Calculate domain and range values to the Vertical stacked bar chart - For Date axis
```

Comment:
how to avoid the code duplication here?

Reply:
Done

Reply:
You can resolve the comments that have been addressed.